mapboxgl/templates/choropleth_alt.html

```diff
@@ -0,0 +1,204 @@
+{% extends "main.html" %}
+
+<!-- update legend item key style -->
+{% block extra_css %}
+<style type='text/css'>
+    .legend div span { border-radius: 20%; }
+</style>
+{% endblock extra_css %}
+
+{% block javascript %}
+var legend = document.getElementById('legend');
+
+mapboxgl.accessToken = '{{ accessToken }}';
+
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: '{{ styleUrl }}', 
+    center: {{ center }},
+    zoom: {{ zoom }},
+    transformRequest: (url, resourceType)=> {
+        if ( url.slice(0,22) == 'https://api.mapbox.com' ) {
+            //Add Python Plugin identifier for Mapbox API traffic
+            return {
+                url: [url.slice(0, url.indexOf("?")+1), "pluginName=PythonMapboxgl&", url.slice(url.indexOf("?")+1)].join('')
+            }
+        }
+        else {
+            //Do not transform URL for non Mapbox GET requests
+            return {url: url}
+        }
+    }
+});
+
+var joinData = {{ joinData }};
+
+map.addControl(new mapboxgl.NavigationControl());
+
+calcCircleColorLegend({{ colorStops }}, "{{ colorProperty }}");
+
+function calcNumericColorValue(num, arr) {
+
+    var curr = arr[0];
+    var diff = Math.abs (num - curr);
+    for (var val = 0; val < arr.length; val++) {
+        var newdiff = Math.abs (num - arr[val]);
+        if (newdiff < diff) {
+            diff = newdiff;
+            curr = arr[val];
+        }
+    }
+    return ({{colorStops}})[curr];
+
+}
+
+map.on('style.load', function() {
+    
+    {% if joinData %}
+        var vectorColorStops = [],
+            popUpKeys = [];
+
+        // create categorical color stops for features in vector layer
+        joinData.forEach(function(row, index) {
+            var red = Math.max(255 - row["{{ colorProperty }}"], 0),
+                green = Math.max(255 - row["{{ colorProperty }}"], 0),
+                blue = Math.min(204 + row["{{ colorProperty }}"], 255);
+
+            var color = "rgba(" + red + ", " + green + ", " + blue + ", 1)";
+            vectorColorStops.push([row["id"], color]);
+            
+            // for adding joined data to pop-up
+            popUpKeys[row["{{ dataJoinProperty }}"]] = row["{{ colorProperty }}"];
+        });
+    {% endif %}
+
+    {% if vectorPolygonSource %}
```

Comment:
@akacarlyann this makes sense to me - define two different source initialization templates, and pick which one to use based on if the source is a vector tile source or geojson polygon source.

Reply:
Sounds good.  Settled on this in b11c006.